msu/utilities/misc.nut

```diff
@@ -1,59 +1,99 @@
 ::MSU.isNull <- function( _object )
 {
-	if (_object == null)
+	return _object == null || (typeof _object == "instance" && _object instanceof ::WeakTableRef && _object.isNull());
+}
+
+::MSU.getField <- function( _key, _object )
```

Comment:
Why are these params in this order?

Reply:
'get this key from that object' makes sense to me. However, it's the other way around for the hook functions, we should probably use that then.

Reply:
This order makes much more sense so I'm really annoyed if we have to end up changing it just because of modhooks. We should encourage the people to use these functions anyway over the modhooks ones (although the modhooks ones are more performant due to avoiding a typeof check in cases where you are sure the object is a table). But yeah I can understand that it will create confusion and likely errors when modders use both MSU and modhooks functions. What do you say we do here then? Use the shit modhooks order?

Reply:
I think it's more intuitive the other way tbh, because to me this replaces `_object._key` but I don't care that much. If you're not convinced just leave it as is and lets go with it.

Reply:
Ok we'll go with `(_object, _key)`

Reply:
Actually on second thought, the idea of these functions is to be used with `::MSU.isIn ( _key, _object )` and there the order is in that order. So it will be a bit weird if someone does:
```squirrel
if (::MSU.isIn(myKey, myObject)
{
   local member = ::MSU.getMember(myObject, myKey) // rather than (myKey, myObject)
}
```
I really can't decide which is better, so @TaroEld and @Enduriel please vote.